### DISCUSSION TITLE
Arguments of Bus class constructor

### DISCUSSION THREAD

**Discussion by BECATRUE** [snu-quiqcl/qiwis#10] [Q&A] [answered]
For implementing Swift class, could you decide arguments of Bus class constructor?

@kangz12345

**Comment by BECATRUE** [comment, marked as answer]
Oh, in the issue #6, there is a documentation.

Sorry..😂

**Reply by kangz12345** [reply]
Ask further questions if there is anything ambiguous!

**Comment by BECATRUE** [comment]
I have a question about the `received` field in `Bus` class!

Why does it have **2** params, not **1** param?  
I think you intend one is a connected `Logic` class, but in `swift`, it will have been connected to appropriate `Logic`s, then isn't it necessary?

@kangz12345

**Reply by kangz12345** [reply]
Good point, 1 parameter is enough.
I have written the detail in #6.

**Reply by BECATRUE** [reply]
Oh, I've been thinking about it again, in order for the destination `Logic` to know which `Bus` the signal came from, the `received` field may need one more param. Otherwise, the information about `Bus` should be appended to `msg`.

I'm sorry to confuse you..

**Reply by kangz12345** [reply]
I think it can be given by `swift`.
In #6, I said like `Bus.received` signal is directly connected to `Logic.received` signal as:

> 7. `Logic` also has `received` signal, and `swift` system has connected the `received` signal of `Bus` to that of the subscriber `Logic` instances of the `Bus`. That is, `bus.received.connect(subscriber.received)` where `subscriber` is a `Logic` instance that subscribes bus.
> 8. It will trigger the connected actions of the subscriber `Logic` instances. Note that since the `Bus.received` signal is emitted by the queue-polling thread, the subscriber's actions may not starts immediately. They will be executed when the control flow gets back to the main thread in which the subscriber `Logic` instance is created.

However, we can make a slot in `swift` so that: `Bus.received` -> the slot emits `Logic.received`.
In this way, `Bus` itself does not have to emit the message with its name since `swift` knows its name.
The slot in `swift` will emit `Logic.received` signal WITH the name of the bus, i.e., its signature is `Logic.received(str, str)`.
Note that this does not directly connect the two signals but 'mediates' them dynamically, which implies flexibility of bus subscription.

**Reply by BECATRUE** [reply]
Umm... It is a good idea that `swift` transfers signals between `Logic` and `Bus`.

However, I think it induces a lot of bottle-nect in terms of `swift`. Thus, I have implemented `swift` only to connect `Logic` to `Bus` or vice versa, not as a moderator of signals.

Also, in order to create a direct bus between two `Logic`s later, I think not using `swift` as a moderator is better.

If you have another opinion, please leave comments.

**Reply by BECATRUE** [reply]
In my method, however, it is inefficient that 1) `swift` can not manage for signal tranfers and 2) `Logic` always emits to all `Bus`es, which I registered candidates of destination.

First of all, I will implement your method!
